pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -0,0 +1,341 @@
+from collections import defaultdict
+from pygen_src.riscv_instr_gen_config import *
+from pygen_src.riscv_instr_pkg import *
+from pygen_src.isa import rv32i_instr
+import random
+from bitstring import BitArray, BitStream
+import logging
+from copy import copy
+class riscv_instr:
+    instr_registry = {}
+    def __init__(self): 
+        self.instr_names = []
+        self.instr_name = None
+        self.instr_group =  defaultdict(list) 
+        self.instr_category = defaultdict(list)
+        self.basic_instr = []
+        self.instr_template = {}
+        
+        self.exclude_reg = []
+        self.include_reg = []
+        
+        self.imm_len = 0
+        self.csr = None
+        self.rs2 = None
+        self.rs1 = None
+        self.rd = None
+        self.imm = BitArray(uint = 0 , length = 32)
+        self.idx = -1
+        self.has_rs1 = 1
+        self.has_rs2 = 1
+        self.has_rd = 1
+        self.has_imm = 1
+
+        self.imm_mask = BitArray(uint = 4294967295 , length = 32)
```

Comment:
For instances like these where variables are initialized to some hex value, you can do this:
```python
self.imm_mask = BitArray(hex="0xffffffff", length=32)
```
for some simplicity